### DISCUSSION TITLE
How do you install the keyboard module with root privileges?

### DISCUSSION THREAD

**Discussion by bryanhughes** [python-poetry/poetry#9258] [General]
Normally, you would do a  `sudo pip install keyboard` for it to gain root permissions. How can I do this is poetry? 

Thanks!
Bryan

**Comment by dimbleby** [comment]
don't do that, use a virtual environment.

**Comment by clintonroy** [comment]
Poetry can't do anything to increase permissions of a program you later run. In general, try to run things with the lowest permissions required. If this is about accessing /dev/input/* under linux, please check the group ownership of them, on my debian box they're group owned by "input", try adding your user to this group.

**Comment by bryanhughes** [comment]
That is what I am asking. The keyboard module requires root access. I do not run my python app as sudo so I am trying to figure out how to pip install keyboard with root permissions.

**Reply by dimbleby** [reply]
> I do not run my python app as sudo so I am trying to figure out how to pip install keyboard with root permissions

This is a misunderstanding.  It does not matter what user installs `keyboard`, it matters what user runs your app.

**Comment by bryanhughes** [comment]
Thank you. I did that:

bryan adm cdrom sudo audio dip video plugdev input render i2c lpadmin gdm sambashare docker weston-launch gpio jtop

Thanks for the help though.
